### PR TITLE
chore(cloudflare): package cleanup

### PR DESCRIPTION
**Pull request by meskill**
**Summary:**  
Activate optimized profile for wasm build

Size dropped from `9,78 MB` to `6,78 MB` for `cloudflare.wasm`

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcall/tree/main/docs

### CODE REVIEW

**Review by ssddOnTop — APPROVED** [review]

**Review by ssddOnTop — COMMENTED** [review]
the total upload size still remains same after the changes

**Comment by meskill** [comment]
> the total upload size still remains same after the changes

is there another tool that makes additional compilation for wasm?

if so then we probably do not need special profiles for cloudflare at all in our config. For now cargo just spams warnings about wrong usage having it inside `cloudflare/Cargo.toml`

**Comment by ssddOnTop** [comment]
We can not do anything externally, wrangler is designed to compile and publish every time, even if we use some external tools, it will just recompile before publishing.

Anyways its nice to save 3 mb of local storage with this pr

**Comment by meskill** [comment]
> We can not do anything externally, wrangler is designed to compile and publish every time, even if we use some external tools, it will just recompile before publishing.
> 
> Anyways its nice to save 3 mb of local storage with this pr

my question was about why the upload size stayed the same after change.

But I think I already got it. It's because wrangler is have step like `[INFO]: Optimizing wasm binaries with wasm-opt` when deploying and therefore those settings are not so important and affects build output only when building purely with cargo

File: cloudflare/package.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "private": true,
   "scripts": {
     "deploy": "npx wrangler deploy --minify",
-    "publish": "npx wrangler publish --minify",
```
Comment (meskill) [inline cloudflare/package.json:7]: publish is deprecated https://developers.cloudflare.com/workers/wrangler/commands/#publish

**Review by ssddOnTop — APPROVED** [review]

**Review by tusharmath — APPROVED** [review]

File: cloudflare/package.json (hunk 1 above)
Comment (tusharmath) [reply cloudflare/package.json:7]: We should add `npm run deploy` so that there is only one command we work with.

File: cloudflare/package.json [hunk 2]
```diff
@@ -4,8 +4,8 @@
...
-    "dev": "npx wrangler dev --port 19194 --remote"
+    "test-debug": "npx wrangler dev --port 19194 --remote --env debug",
+    "test-release": "npx wrangler dev --port 19194 --remote"
```
Comment (tusharmath) [inline cloudflare/package.json:8]:
```suggestion
    "dev-debug": "npx wrangler dev --port 19194 --env debug",
    "dev": "npx wrangler dev --port 19194"
```

This is not a test command so let's not use `test`. The additional params such as `--remote` can be passed anyways with `npm run` so can drop those too.

File: cloudflare/package.json (hunk 1 above)
Comment (meskill) [reply cloudflare/package.json:7]: it's right above it

**Review by tusharmath — APPROVED** [review]

**Review by tusharmath — APPROVED** [review]

**Pull request merged**